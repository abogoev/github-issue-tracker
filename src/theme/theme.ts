import { TextStyle } from "react-native";

interface Theme {
  palette: typeof palette;
  typography: {
    verylargeText: TextStyle;
    largeText: TextStyle;
    regularText: TextStyle;
    regularTextBold: TextStyle;
    smallText: TextStyle;
  };
  borderRadius: 6;
}

const HELVETICA = "Helvetica";

const palette = {
  green: "rgb(45, 164, 78)",
  blue: "rgb(9, 105, 218)",
  red: "rgb(207, 34, 46)",
  purple: "rgb(130, 80, 223)",
  darkBlue: "rgba(9,105,218,0.3)",
  white: "#fff",
  transparent: "transparent",
  lightGrey: "rgb(246, 248, 250)",
  grey: "rgb(216, 222, 228)",
  neutral: "rgb(208, 215, 222)",
  darkGrey: "rgb(87, 96, 106)",
  veryDarkGrey: "#24292f",
} as const;

const theme: Theme = {
  palette,
  typography: {
    verylargeText: {
      fontFamily: HELVETICA,
      fontSize: 24,
      lineHeight: 36,
      fontWeight: "300",
    },
    largeText: {
      fontFamily: HELVETICA,
      fontSize: 16,
      lineHeight: 24,
      fontWeight: "600",
    },
    regularText: {
      fontFamily: HELVETICA,
      fontSize: 14,
      lineHeight: 21,
      fontWeight: "400",
    },
    regularTextBold: {
      fontFamily: HELVETICA,
      fontSize: 14,
      lineHeight: 20,
      fontWeight: "500",
    },
    smallText: {
      fontFamily: HELVETICA,
      fontSize: 12,
      lineHeight: 18,
    },
  },
  borderRadius: 6,
};

export default theme;
