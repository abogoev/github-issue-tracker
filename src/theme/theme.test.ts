import theme from "./theme";

describe("theme", () => {
  it("should match snapshot", () => {
    expect(theme).toMatchSnapshot();
  });
});
