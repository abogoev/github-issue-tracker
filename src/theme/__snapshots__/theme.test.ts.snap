// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme should match snapshot 1`] = `
Object {
  "borderRadius": 6,
  "palette": Object {
    "blue": "rgb(9, 105, 218)",
    "darkBlue": "rgba(9,105,218,0.3)",
    "darkGrey": "rgb(87, 96, 106)",
    "green": "rgb(45, 164, 78)",
    "grey": "rgb(216, 222, 228)",
    "lightGrey": "rgb(246, 248, 250)",
    "neutral": "rgb(208, 215, 222)",
    "purple": "rgb(130, 80, 223)",
    "transparent": "transparent",
    "veryDarkGrey": "#24292f",
    "white": "#fff",
  },
  "typography": Object {
    "largeText": Object {
      "fontFamily": "Helvetica",
      "fontSize": 16,
      "fontWeight": "600",
      "lineHeight": 24,
    },
    "regularText": Object {
      "fontFamily": "Helvetica",
      "fontSize": 14,
      "fontWeight": "400",
      "lineHeight": 21,
    },
    "regularTextBold": Object {
      "fontFamily": "Helvetica",
      "fontSize": 14,
      "fontWeight": "500",
      "lineHeight": 20,
    },
    "smallText": Object {
      "fontFamily": "Helvetica",
      "fontSize": 12,
      "lineHeight": 18,
    },
    "verylargeText": Object {
      "fontFamily": "Helvetica",
      "fontSize": 24,
      "fontWeight": "300",
      "lineHeight": 36,
    },
  },
}
`;
